### PR TITLE
[Refactor] Remove unnecessary null checks

### PR DESCRIPTION
**Pull request by sduzh**
- Remove unnecessary null checks
- Add some NotNull annotations

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 4 / 4 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|4|4|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/transaction/GlobalTransactionMgr.java [hunk 1]
```diff
@@ -359,17 +361,22 @@ public TransactionState getLabelTransactionState(long dbId, String label) {
...
+    @NotNull
+    public VisibleStateWaiter commitTransaction(long dbId, long transactionId,
+                                                @NotNull List<TabletCommitInfo> tabletCommitInfos,
                                                 TxnCommitAttachment txnCommitAttachment)
             throws UserException {
         return commitTransaction(dbId, transactionId, tabletCommitInfos, Lists.newArrayList(), txnCommitAttachment);
     }
 
+    @NotNull
     public VisibleStateWaiter commitTransaction(long dbId, long transactionId, List<TabletCommitInfo> tabletCommitInfos,
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/transaction/GlobalTransactionMgr.java:380]:
```suggestion
    public VisibleStateWaiter commitTransaction(long dbId, long transactionId, @NotNull List<TabletCommitInfo> tabletCommitInfos,
```

Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/transaction/GlobalTransactionMgr.java]: Updated

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 4 / 4 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|4|4|100.00%|[]|

**Review by gengjun-git — APPROVED** [review]

**Comment by sduzh** [comment]
@Mergifyio backport branch-3.1

**Comment by sduzh** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**